### ISSUE TITLE
Repeatead chars not working as expected

### ISSUE THREAD

**Issue by esfomeado** [jqwik-team/jqwik#506]
## Testing Problem

I had an issue where jqwik is generating duplicated values frequently which, for my example should almost never occur.

I have noticed that these duplicated values are always the same character e,g, `AAAAAA` so I decided to set the `repeatedChars` to 0 but this type of values are still generated. I have tried the other values but seems that there is no impact on the ammount of time a repeated char is used.

```
    @Test
    void bug() {
        StringArbitrary strings = Arbitraries.strings()
                .alpha()
                .ofMinLength(5)
                .ofMaxLength(25)
                .repeatChars(0);

        List<String> names = IntStream.range(0, 100)
                .mapToObj(i -> strings.sample())
                .toList();

        Map<String, Long> duplicateCount = names.stream()
                .filter(Objects::nonNull)
                .collect(Collectors.groupingBy(s -> s, Collectors.counting()));

        List<String> duplicateStrings = duplicateCount.entrySet().stream()
                .filter(entry -> entry.getValue() > 1)
                .map(Map.Entry::getKey)
                .collect(Collectors.toList());

        System.out.println("Duplicate strings: " + duplicateStrings);

        assertThat(duplicateStrings).isEmpty();
    }
```

**Comment by jlink** [comment]
@esfomeado Thanks for raising that issue. 
I see why you‘d expect that `repeatChars(0)` to not create any duplicate chars at all. Currently it just sets the probability of _explicitly adding_ duplicate chars to 0. The effect you see is due to edge case resolution. 

I’ll have to think about if changing the behaviour or documentation of `repeatChars` is the right thing to do. Maybe an additional `uniqueChars()` would be even clearer. 

To solve your problem short-term, you can use [withoutEdgeCases()](https://jqwik.net/docs/1.7.4/javadoc/net/jqwik/api/Arbitrary.html#withoutEdgeCases()) on the strings arbitrary, which will prevent „AAAAA“ to be generated on purpose. Duplicate chars will still occur by chance though. 

If you definitely want no duplicates, you’d currently have to take a detour through something like:
`Arbitraries.chars().list().ofMaxSize(25).uniqueElements().map(…)`

**Comment by esfomeado** [comment]
Thanks for the quick response.

@jlink Personally I would expect to have the same behaviour was `uniqueChars` when set to 0 so I would say to just call `uniqueChars`.

**Comment by jlink** [comment]
The `StringArbitrary.uniqueChars()` functionality and the fix for `repeatChars(0.0)` is available in `1.8.0-SNAPSHOT`.

**Comment by jlink** [comment]
In addition, there's now an annotation `@UniqueChars` that can be applied to any `String` parameter in a property.

**Comment by jlink** [comment]
New features available in 1.8.0-SNAPSHOT
